src/query/service/src/servers/http/v1/query/execute_state.rs

```diff
@@ -112,6 +112,7 @@ pub struct ExecuteStopped {
     pub affect: Option<QueryAffect>,
     pub reason: Result<()>,
     pub stop_time: Instant,
+    pub session: Arc<Session>,
```

Comment:
not very sure it is ok,  not releasing session as fast as possible may lead to unexpected behavior.

Reply:
or even problems for shutdown? cc @zhang2014

Reply:
I suppose this might not extend the lifetime much longer. The session object will be recycled when the ref count reaches zero, and it's still very fast to release this ref count on the response.

Reply:
```
pub struct HttpQuery {
    pub(crate) id: String,
    pub(crate) session_id: String,
    request: HttpQueryRequest,
    state: Arc<RwLock<Executor>>,
```

Executor not released until HttpQuery is cleared by final or timeout